Multihost command support for mininet CLI

fetzerms opened the pull request:
This patch adds the possiblity to execute commands on multiple hosts using one command. This is similar to the existing scheme, for single hosts like: <hostname> <command>. For example: "h1 uptime".

This patch allows to specific ranges and lists of hosts. See the examples below:

Using a Range:
```
mininet> h[1-2] uptime
*** Executing command on node: h1
 16:49:47 up  2:05,  7 users,  load average: 0.56, 0.48, 0.58
*** Executing command on node: h2
 16:49:47 up  2:05,  7 users,  load average: 0.56, 0.48, 0.58
```

Using a comma seperated list:
```
mininet> h[1,2] uptime
*** Executing command on node: h1
 16:50:42 up  2:06,  7 users,  load average: 0.50, 0.48, 0.58
*** Executing command on node: h2
 16:50:42 up  2:06,  7 users,  load average: 0.50, 0.48, 0.58
```

Using a combination of both:
```
mininet> h[1,3-4] uptime
*** Executing command on node: h1
 16:51:28 up  2:07,  7 users,  load average: 0.27, 0.42, 0.55
*** Executing command on node: h3
 16:51:28 up  2:07,  7 users,  load average: 0.27, 0.42, 0.55
*** Executing command on node: h4
 16:51:28 up  2:07,  7 users,  load average: 0.27, 0.42, 0.55
```

I tested this during the experiments of my thesis with up to 100 hosts, and it worked flawless there. If the pull request has a change to be accepted, I can update the documentation as well.

Any comments or suggestions are welcome.

fetzerms commented:
Any comments on this one? Any special fixes / updates needed to be accepted?

fetzerms commented:
Is this PR worth updating, or won't it be merged/considered anyways?

lantz commented:
I actually like this feature - it seems useful and consistent with the mininet idea of "one console to rule them all."

fetzerms commented:
Ok, I will update my PR and some documentation about it during this week.

fetzerms commented:
Updated the PR. Please review.

fetzerms commented:
@lantz Any update on this one?

lantz commented:
@fetzerms So a few things:

1. This seems to break lines like
    mininet> h1 if [ -e /etc/passwd ]; then echo "great!"; else echo "failure"; fi
2. It might make sense to make a helper function
3. The code doesn't seem to match the [Mininet Python Style](https://github.com/mininet/mininet/wiki/Mininet-Python-Style)

fetzerms commented:
Ok, I will look into it. What do you suggest for 1)? Check if the brackets occour in the first word?
I can move it to a helper function and follow the python style. Tho my python skills are rather mediocre.

fetzerms commented:
@lantz I fixed the breaking of lines containing brackets and moved the parsing of hosts to a helper function. I also ensured to use camel case for my variables.

lantz commented:
```
mininet> h[x] hostname
--------------------------------------------------------------------------------
Caught exception. Cleaning up...

ValueError: invalid literal for int() with base 10: 'x'
```
